test/rewards/AutoCompoundingRewards.ts

```diff
@@ -686,6 +711,83 @@ describe('AutoCompoundingRewards', () => {
+                                distributionType,
+                                autoCompoundingRewards,
+                                tokens[index].address,
+                                setup.totalRewards,
+                                START_TIME
+                            );
+                        }
+
+                        await autoCompoundingRewards.setAutoTriggerCount(AUTO_TRIGGER_COUNT);
+                    });
+
+                    if (distributionType === RewardsDistributionType.Flat) {
+                        it('should distribute all tokens', async () => {
+                            await autoCompoundingRewards.setTime(programEndTime[distributionType]);
+                            for (let i = 0; i < Math.ceil(setups.length / AUTO_TRIGGER_COUNT); i++) {
+                                const res = await autoCompoundingRewards.trigger();
+                                await expect(res).to.emit(autoCompoundingRewards, 'RewardsDistributed');
+                            }
+                            const res = await autoCompoundingRewards.trigger();
+                            await expect(res).not.to.emit(autoCompoundingRewards, 'RewardsDistributed');
+                        });
+
+                        it('should distribute some tokens', async () => {
+                            for (let i = 0; i < 5; i++) {
+                                await autoCompoundingRewards.setTime(
+                                    Math.floor(START_TIME + (programDuration[distributionType] * 2 ** i) / (2 ** i + 1))
+                                );
+                                const res = await autoCompoundingRewards.trigger();
+                                await expect(res).to.emit(autoCompoundingRewards, 'RewardsDistributed');
+                            }
+                        });
+
+                        it('should not distribute tokens again after less than 1 hour', async () => {
+                            await autoCompoundingRewards.setTime(
+                                Math.floor(START_TIME + programDuration[distributionType] / 2)
+                            );
+                            for (let i = 0; i < Math.ceil(setups.length / AUTO_TRIGGER_COUNT); i++) {
+                                const res = await autoCompoundingRewards.trigger();
+                                await expect(res).to.emit(autoCompoundingRewards, 'RewardsDistributed');
+                            }
+                            await autoCompoundingRewards.setTime(
+                                Math.floor(START_TIME + programDuration[distributionType] / 2) + 60 * 60 - 1
+                            );
+                            for (let i = 0; i < Math.ceil(setups.length / AUTO_TRIGGER_COUNT) + 1; i++) {
+                                const res = await autoCompoundingRewards.trigger();
+                                await expect(res).not.to.emit(autoCompoundingRewards, 'RewardsDistributed');
```

Comment:
Can you add a test that only some programs are skipped and not only them? You can add a `TestAutoCompoundingRewards`-only function that allows you to override `prevDistributionTimestamp` (or to manually create a few "holes" by manually triggering their rewards processing)

Reply:
It's already there (one test above this one).